packages/api/cms-api/src/dependencies/dependencies.service.ts

```diff
@@ -77,19 +76,71 @@ export class DependenciesService {
         console.timeEnd("refresh materialized block dependency");
     }
 
-    async getDependents(target: AnyEntity<{ id: string }>): Promise<Dependency[]> {
-        const entityName = target.constructor.name;
-        return this.connection.execute(`SELECT * FROM block_index_dependencies as idx WHERE idx."targetEntityName" = ? AND idx."targetId" = ?`, [
-            entityName,
-            target.id,
-        ]);
+    async getDependents(
+        target: AnyEntity<{ id: string }> | { entityName: string; id: string },
+        filter?: {
+            rootGraphqlObjectType?: string;
+            rootEntityName?: string;
+            rootId?: string;
+            rootColumnName?: string;
+        },
+    ): Promise<Dependency[]> {
+        await this.refreshViews();
```

Comment:
If you always refresh views we should not have materialized views at all.

Instead we could:
- refresh after a period of time (eg. 15 minutes or 1 hour)
- refresh in background (like stale while revalidate)
- in the dependencies views offer the user a refesh now button to trigger the refresh
- for other usages (eg. checking dependencies when deleting a dam file) we _need_ up to date data, we need a Api requesting that (eg as argument to the FieldResolver)

Reply:
That's what I implemented here: https://github.com/vivid-planet/comet/pull/1222 

It's the same method, so I didn't make these PRs depending on one another

Reply:
oh, sorry, somehow I completely forgot about that